substrate/frame/message-queue/src/lib.rs

```diff
@@ -1554,12 +1564,17 @@ impl<T: Get<O>, O: Into<u32>> Get<u32> for IntoU32<T, O> {
 impl<T: Config> ServiceQueues for Pallet<T> {
 	type OverweightMessageAddress = (MessageOriginOf<T>, PageIndex, T::Size);
 
-	fn service_queues(weight_limit: Weight) -> Weight {
+	fn service_queues(weight_limit: Weight, context: ServiceQueuesContext) -> Weight {
```

Comment:
I just noticed that this is implementing a trait function. Can you maybe put the inner logic into a new function and call that from here?  

Having the `ServiceQueuesContext` in the interface would be ugly otherwise.

Reply:
Thanks, I pushed a new commit last night (PST). Now that I'm re-reading this I might not have implemented the way you requested.

Reply:
I mean, the code does not compile? You are implementing a trait function. Please try to run the pallet tests locally.